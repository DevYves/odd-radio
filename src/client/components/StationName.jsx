import React, {Component} from 'react';

class StationName extends Component {
  render(){


    return(

        <div className="player-station-name"> { this.props.StationFeed } </div>

      );
  }
}
export default StationName;