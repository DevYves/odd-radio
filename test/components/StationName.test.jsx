import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import StationName from '../../src/client/components/StationName';

describe('StationName Component', () => {
  const props = {
    Style: 'test-style',
    StationFeed: jest.fn(),
    findStationExpandInfo: jest.fn()
  }

  const subject = <StationName { ...props}/>;

  it('displays div with className = this.props.style', () => {
      const wrapper = shallow(subject);

      expect(wrapper.find('.test-style').length).toBe(1);
  });

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('When clicked on', () => {
    const wrapper = shallow(subject);

    wrapper.find('.test-style').simulate('click');

    expect(props.findStationExpandInfo).toHaveBeenCalled();
  })

});

