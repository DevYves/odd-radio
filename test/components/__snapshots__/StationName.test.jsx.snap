// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationName Component matches snapshot 1`] = `
<div
  className="test-style"
  onClick={[Function]}
>
   
   
</div>
`;
